docs/sources/flow/setup/deploy-agent.md

```diff
@@ -0,0 +1,118 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/setup/deploy-agent/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/setup/deploy-agent/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/setup/deploy-agent/
+canonical: https://grafana.com/docs/agent/latest/flow/setup/start-agent/
+description: Learn about possible deployment topologies for Grafana Agent
+menuTitle: Deploying Grafana Agent
+title: Grafana Agent deployment topologies
+weight: 900
+---
+
+# Deploying Grafana Agent
+
+Grafana Agent is a flexible, vendor-neutral telemetry collector. This
+flexibility means that the Agent doesn’t enforce a specific deployment topology
+but can work in multiple scenarios.
+
+This page lists common topologies used for deployments of Grafana Agent, when
+to consider using each, issues that users may run into, as well as scaling
+considerations.
+
+## As a centralized collection service
+Deploying the Grafana Agent as a centralized service is recommended for
+collecting application telemetry. This allows a smaller set of agents to
+coordinate service discovery, collection, and remote writing.
+
+Using this topology requires deploying the Agent on separate infrastructure,
+and making sure that agents can discover and reach these applications over the
+network. The main predictor for the size of the agent is the number of active
+metrics series it is scraping; a rule of thumb is ~10kb of memory for each
+series. We recommend you start looking towards horizontal scaling around the 1M
+active series mark.
+
+### Using Kubernetes StatefulSets
+Deploying Grafana Agent as a StatefulSet is the recommended option for metrics
+collection.
+The persistent pod identifiers make it possible to consistently match volumes
+with pods so that you can use them for the WAL directory.
+
+You can also use a Kubernetes deployment in cases where persistent storage is not required, such as a traces-only pipeline.
+
+### Pros
```

Comment:
looks like the heading level matches `### Using Kubernetes StatefulSets`. is this a subsection and should have 4 #s? Similar question throughout but I won't open a comment for each.

Reply:
It's not supposed to be a subsection of the Kubernetes deployment, but rather the idea was the following structure.

* X
* X on Kubernetes
* Pros of X
* Cons of X
* etc...

If it's not apparent I can move things around for the K8S section to be at the bottom.

Reply:
I ran `make docs` on Paschalis' branch, and I think the headings are much more clear over there. So I think it's ok to just leave it as it is.